services/CustomFees.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "BasicTypes.proto";
+
+/* A fraction of the transferred units of a token to assess as a fee. The amount assessed
+will never be less than the given minimum_amount, and never greater than the given maximum_amount. 
+The denomination is always units of the token to which this fractional fee is attached. */
+message FractionalFee {
+  Fraction fractional_amount = 1; // The fraction of the transferred units to assess as a fee
+  sint64 minimum_amount = 2; // The minimum amount to assess
+  sint64 maximum_amount = 3; // The maximum amount to assess (zero implies no maximum)
+}
+
+/* A fixed number of units (hbar or token) to asssess as a fee during a CryptoTransfer 
+that transfers units of the token to which this fixed fee is attached. */
+message FixedFee {
+  sint64 amount = 1; // The number of units to assess as a fee
```

Comment:
Q: can `amount` be negative? If not, `sint64` does not provide benefit and `int64` is more appropriate. The same applies to most other `sint64` fields.

Reply:
Since Java does not have an unsigned integral type, we can't actually support the full range of values `[0, 2^64 - 1]`, so this is a more honest contract.

Reply:
understood. `int64` and `sint64` are both signed, the difference as in protobuf afaik is `sint64` is encoded in a zigzag way which reduces encoded message size for negative value. so if `amount` is always non-negative, `int64` suites better v.s. `sint64`.

Reply:
Ah I see, changed all `sint64` to `int64`. 👍